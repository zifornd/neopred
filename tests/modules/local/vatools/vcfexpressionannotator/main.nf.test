nextflow_process {

    name "Test Process VATOOLS_VCFEXPRESSIONANNOTATOR"
    script "modules/local/vatools/vcfexpressionannotator/main.nf"
    process "VATOOLS_VCFEXPRESSIONANNOTATOR"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
		input[0] = Channel.of([
                    [ id:'SRR8281226' ], 
                    [ file("$projectDir/tests/test_result/gunzip/SRR8281226.annotated.vcf", checkIfExists: true) ]
                ])

		   input[1] = file("$projectDir/tests/test_result/batch_removal/design_batch_tpm.genesymbol.batchremoved.csv")

                """
            }
        }

        then {
            assert process.success 
	    assert process.out.expr_vcf
	    assert process.out.versions
            assert snapshot(process.out).match()
        }

    }

}
