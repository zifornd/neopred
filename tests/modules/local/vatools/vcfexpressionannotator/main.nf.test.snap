{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mutect2.somatic.base.snp.Somatic.hc.filter.vep.gx.vcf:md5,6faa888d8f2cc3953b5f04b1c74ba687"
                    ]
                ],
                "1": [
                    "versions.yml:md5,75870ffa0a81b1bf39d09f485c4fd990"
                ],
                "expr_vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mutect2.somatic.base.snp.Somatic.hc.filter.vep.gx.vcf:md5,6faa888d8f2cc3953b5f04b1c74ba687"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,75870ffa0a81b1bf39d09f485c4fd990"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2025-01-08T06:18:47.046469"
    }
}